src/core/json/driver.h

```diff
@@ -6,24 +6,85 @@
 
 #include <memory>
 #include <string>
+#include <vector>
 
 namespace dfly {
 namespace json {
 
 class Lexer;
+class location;  // from jsonpath_grammar.hh
+
+enum class SegmentType {
+  IDENTIFIER = 1,  // $.identifier
+  INDEX = 2,       // $.array[0]
+  WILDCARD = 3,    // $.array[*] or $.*
+};
+
+class PathSegment {
+ public:
+  PathSegment(SegmentType type, std::string identifier = std::string())
+      : type_(type), identifier_(std::move(identifier)) {
+  }
+
+  SegmentType type() const {
+    return type_;
+  }
+
+  const std::string& identifier() const {
+    return identifier_;
+  }
+
+ private:
+  SegmentType type_;
+  std::string identifier_;
+  int index_;
+};
+
+class Path {
```

Comment:
I don't understand why you need this class. It duplicates std::vector<PathSegment> interface

Reply:
it's gonna grow in the future. the code is not finished yet